src/Symfony/Component/Security/Core/Authentication/Provider/LdapBindAuthenticationProvider.php

```diff
@@ -40,11 +40,11 @@ class LdapBindAuthenticationProvider extends UserAuthenticationProvider
      * @param UserProviderInterface $userProvider               A UserProvider
      * @param UserCheckerInterface  $userChecker                A UserChecker
      * @param string                $providerKey                The provider key
-     * @param LdapClientInterface   $ldap                       An Ldap client
+     * @param LdapInterface         $ldap                       A Ldap client
      * @param string                $dnString                   A string used to create the bind DN
      * @param bool                  $hideUserNotFoundExceptions Whether to hide user not found exception or not
      */
-    public function __construct(UserProviderInterface $userProvider, UserCheckerInterface $userChecker, $providerKey, LdapClientInterface $ldap, $dnString = '{username}', $hideUserNotFoundExceptions = true)
+    public function __construct(UserProviderInterface $userProvider, UserCheckerInterface $userChecker, $providerKey, LdapInterface $ldap, $dnString = '{username}', $hideUserNotFoundExceptions = true)
```

Comment:
This is a BC break as you now cannot pass `LdapClientInterface` implementations anymore.